dev/puppet-server.conf.sample

```diff
@@ -115,3 +115,8 @@ authorization: {
         }
     ]
 }
+
+# general puppetserver settings
+puppetserver: {
+    environment-class-cache-enabled: true
+}
```

Comment:
I was thinking of adding in commented out settings for the `versioned-code` block (which I think is what Kevin was referring to as being in a `puppetserver` block, except they turn out to actually not be) to this sample config file, which is our main "dev" example one. I don't think it's packaged, and we already have a lot of comments in it, so that seems okay?